cmd/prepare-storage.go

```diff
@@ -142,12 +143,19 @@ func prepForInitXL(firstDisk bool, sErrs []error, diskCount int) InitActions {
 		return WaitForQuorum
 	}
 
-	// There is quorum or more corrupted disks, there is not enough good
-	// disks to reconstruct format.json.
+	// There is quorum or more corrupted disks, there is not
+	// enough good disks to reconstruct format.json.
 	if disksCorrupted >= quorum {
 		return Abort
 	}
 
+	// If we have a combination of disks corrupted, disks unformatted
+	// equal to read quorum and disks formatted are lesser than read
+	// quorum we should abort.
+	if disksCorrupted+disksUnformatted == readQuorum && disksFormatted < readQuorum {
```

Comment:
With heal-format command becoming available, we should abort only when `disksFormatted + disksOffline < readQuorum`. I.e, optimistically assuming that all disksOffline are in fact formatted disks. For all other configurations, it is worth waiting for a heal-format command being run in future. Thoughts?